### DISCUSSION TITLE
Using `-si 0.5` doesn't seem to simplify the meshes at all

### DISCUSSION THREAD

**Discussion by sotrh** [zeux/meshoptimizer#749] [Q&A] [answered]
I'm looking to implement a LOD system using meshoptimizer/gltfpack and for each LOD I specify distance value and a simplification threshold that I pass to gltfpack. The meshes this generates seem to be the same as the ones with no threshold supplied. I imagine that there's probably some extra flag I need to set.

**Comment by zeux** [comment, marked as answer]
`-si 0.5` specifies the *target* simplification ratio. However, several additional factors may restrict simplification in practice:

1. Simplification respects mesh topology by default and may refuse to simplify areas where the vertices are duplicated too often; an extreme example of this is a faceted model (eg a sphere with flat shading) where the default simplifier has no room to collapse anything because it respects attribute seams. You can usually tell at a glance if this will be a problem by comparing the number of triangles vs number of vertices in mesh primitive stats in verbose (`-v`) output; the closer number of vertices is to number of triangles times three, the more it is a problem (even a factor of 2 is highly problematic as it suggests meshes built out of faceted quads).
2. In gltfpack, simplification deviation is currently hardcoded and defaults to ~1% deviation; this is tracked in https://github.com/zeux/meshoptimizer/issues/494 to expose a command line option for, but this means a low-poly mesh may not have any freedom in simplification by default.
3. Finally, a variant of (1) sometimes occurs on some glTF files in the wild where the number of vertices is close to the number of triangles times three but the mesh does not appear faceted; I've seen this occasionally on some files converted from FBX and/or some Sketchfab models, where tolerance based welding would fix the problem. gltfpack will probably start doing that using the existing settings (eg based on whether or not normals are quantized) in the future.

In addition to understanding *why* simplification doesn't simplify based on above, you can try `-sa` option (this is documented in full gltfpack help, `gltfpack -h`, which I recommend reading) which ignores topological restrictions (1, 3) and uses a much larger deviation limit (2) -- this may result in significant quality degradation depending on the input model, but should achieve the target ratios.

**Reply by sotrh** [reply]
Ok thanks. I was wondering about `-sa`. I can use both `-sa` and `-si` right?
